packages/esm-patient-labs-app/src/lab-orders/add-lab-order/lab-order-form.component.tsx

```diff
@@ -13,31 +13,37 @@ import {
   Grid,
   InlineNotification,
   TextInput,
-  FormLabel,
+  TextArea,
 } from '@carbon/react';
 import { useTranslation } from 'react-i18next';
 import { priorityOptions } from './lab-order';
 import { useTestTypes } from './useTestTypes';
 import styles from './lab-order-form.scss';
-import { Controller, useForm } from 'react-hook-form';
+import { Controller, type FieldErrors, useForm } from 'react-hook-form';
 import { zodResolver } from '@hookform/resolvers/zod';
 import { z } from 'zod';
+import { moduleName } from '@openmrs/esm-patient-chart-app/src/constants';
 
 export interface LabOrderFormProps {
   initialOrder: LabOrderBasketItem;
   closeWorkspace: () => void;
 }
 
 const labOrderFormSchema = z.object({
-  urgency: z.string({ required_error: 'Priority is required', invalid_type_error: 'Priority is required' }),
+  urgency: z.string({
+    required_error: translateFrom(moduleName, 'addLabOrderPriorityRequired', 'Priority is required'),
+    invalid_type_error: translateFrom(moduleName, 'addLabOrderPriorityRequired', 'Priority is required'),
+  }),
   instructions: z.string().optional(),
-  labReferenceNumber: z.string({
-    required_error: 'Lab reference number is required',
-    invalid_type_error: 'Lab reference number is required',
+  labReferenceNumber: z.string().refine((value) => value !== '', {
+    message: translateFrom(moduleName, 'addLabOrderLabReferenceRequired', 'Lab reference number is required'),
   }),
   testType: z.object(
     { label: z.string(), conceptUuid: z.string() },
-    { required_error: 'Test type is required', invalid_type_error: 'Test type is required' },
+    {
+      required_error: translateFrom(moduleName, 'addLabOrderLabTestTypeRequired', 'Test type is required'),
+      invalid_type_error: translateFrom(moduleName, 'addLabOrderLabReferenceRequired', 'Test type is required'),
```

Comment:
Do we need both `invalid_type_error` and `required_error`?

Reply:
For object it is need as when their is no option selected, it has `null` as value thus the the type error
cc @denniskigen  @vasharma05

Reply:
This code is for a string type